src/v/kafka/server/connection_context.h

```diff
@@ -100,7 +100,9 @@ class connection_context final
 
     protocol& server() { return _proto; }
     const ss::sstring& listener() const { return _rs.conn->name(); }
-    security::sasl_server& sasl() { return _sasl; }
+    security::sasl_server* sasl() {
+        return _sasl.has_value() ? &_sasl.value() : nullptr;
+    }
```

Comment:
I don't get why this has to return a pointer. Why not just use `sasl.value()`?

Reply:
`sasl` is an `optional<sasl_server>` that isn't always engaged.